### PR TITLE
chore(deps): Updating FEC and related packages

### PR DESCRIPTION
**Pull request by Andrewgdewar**
# Description

This: 
- updates a number of modules identified by snyk
- removes many unneeded babel artifacts
- updates test configurations
- updates the fec configuration file

Why: 
- I am running a macbook and thus use Docker, this application used an older fec-config module that had a bug preventing usage with docker.
- This fec-config bump necessitated a number of other updates, although some are not required, these changes help to simplify/modernify this application.

# How to test the PR

Please include steps to test your PR.

# After the change
- Sanity check, make sure all things still work as expected. 

# Checklist:

- [x] The commit message has the Jira ticket linked
- [x] PR has a short description
~~- [ ] Screenshots before and after the change are added~~
- [x] Pre-existing tests are still passing  ~~for the changes have been added~~
- [x] README.md is updated if necessary
~~- [ ] Needs additional dependent work~~

### CODE REVIEW

File: fec.config.js [hunk 1]
```diff
@@ -1,27 +1,56 @@
...
-                    version: '^6.8.1',
-                    requiredVersion: '>=6.0.0 <7.0.0'
-                }
-            }
+                    version: dependencies["react-router-dom"],
+                    requiredVersion: ">=6.0.0 <7.0.0",
+                },
+            },
+            {
+                "@unleash/proxy-client-react": {
```
Comment (bastilian) [inline fec.config.js:33]: This shouldn't be needed, and might actually cause issues. Chrome is already sharing its instance of the unleash client.

File: fec.config.js [hunk 2]
```diff
@@ -1,27 +1,56 @@
...
-        exposes: {
-            './RootApp': resolve(__dirname, './src/AppEntry'),
-            './SystemDetail': resolve(__dirname, './src/index.js')
-        }
-    }
+    },
+    /**
+     * Add additional webpack plugins
+     */
+    resolve: {
```
Comment (bastilian) [inline fec.config.js:43]: Why is this needed?

File: fec.config.js (hunk 1 above)
Comment (Andrewgdewar) [reply fec.config.js:33]: Good call, this and the url below are left-overs from a copy of code 🫤.

This isn't ready for a review.

File: fec.config.js (hunk 2 above)
Comment (Andrewgdewar) [reply fec.config.js:43]: Maybe this is handled by chroming, if so it can be removed. 
My understanding is it tells the parent app where our node_modules folder is. 
It was included in the template code I used for my application.
@Hyperkid123?

File: fec.config.js [hunk 3]
```diff
@@ -1,27 +1,31 @@
...
             {
-                'react-router-dom': {
+                "react-router-dom": {
                     singleton: true,
                     import: false,
-                    version: '^6.8.1',
-                    requiredVersion: '>=6.0.0 <7.0.0'
-                }
-            }
+                    version: dependencies["react-router-dom"],
```
Comment (adonispuente) [inline fec.config.js:25]: I believe this needs to be `version: packageJson.dependencies['react-router-dom'],`

**Review by adonispuente — COMMENTED** [review]
The app itself seems to start up and work as it should, just left a comment about the fec.config.

File: package.json [hunk 4]
```diff
@@ -6,71 +6,43 @@
...
-      "\\.(css|scss)$": "identity-obj-proxy"
-    },
-    "testPathIgnorePatterns": [
-      "src/SmartComponents/SystemDetail/InventoryDetail.test.js",
-      "src/SmartComponents/PatchSet/PatchSet.test.js"
-    ],
-    "transformIgnorePatterns": [
-      "<rootDir>/node_modules/(?!(@patternfly/react-core/|@patternfly/react-icons/|@redhat-cloud-services|@openshift|lodash-es|@patternfly/react-table|@patternfly/react-tokens|p-all)).*$"
-    ]
-  },
```
Comment (Andrewgdewar) [inline package.json:56]: I just moved this to it's own file and added a single transform. Easier to track changes.

File: babel.config.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{
+  "presets": ["@babel/preset-react", "@babel/preset-env"],
```
Comment (Andrewgdewar) [inline babel.config.json:2]: The below is largely the same, except for the removal of two babel plugins.  
The file type change was required due to updates.

**Review by adonispuente — COMMENTED** [review]
LGTM. Though because of the amount of packages changing id rather have at least 1 other person review the changes. 

Overall I havent found anything concerning. The title says youre updating FEC config FOR the ticket, but id rather it just be ‘updating FEC and related packages’ in the commit message/title. I dont see how the two items are related. This is just for cleaner commit history.

File: fec.config.js (hunk 2 above)
Comment (Hyperkid123) [reply fec.config.js]: Unless your project resolves modules from a different directory, this is not needed.

File: fec.config.js [hunk 6]
```diff
@@ -1,27 +1,31 @@
-const { resolve } = require('path');
+const path = require("path");
+const { dependencies, insights } = require("./package.json");
+
+const moduleName = insights.appname.replace(/-(\w)/g, (_, match) =>
+    match.toUpperCase()
+);
```
Comment (AsToNlele) [inline fec.config.js:6]: why not just use `insights.appname`?

File: fec.config.js [hunk 7]
```diff
@@ -1,27 +1,31 @@
...
+
+const srcDir = path.resolve(__dirname, "./src");
```
Comment (AsToNlele) [inline fec.config.js:8]: this is not used afaik?

File: fec.config.js [hunk 8]
```diff
@@ -1,27 +1,31 @@
...
-    useProxy: process.env.PROXY === 'true',
-    proxyVerbose: true,
-    plugins: [],
-    ...(process.env.HOT ? { hotReload: process.env.HOT === 'true' } : { hotReload: true }),
-    ...(process.env.port ? { port: parseInt(process.env.port) } : {}),
+    appName: moduleName,
+    appUrl: "/insights/patch",
+    useProxy: process.env.PROXY === "true",
     moduleFederation: {
+        moduleName,
```
Comment (AsToNlele) [inline fec.config.js:15]: I don't see this eg in compliance, is this needed?

**Review by AsToNlele — COMMENTED** [review]
Left some questions

File: fec.config.js (hunk 8 above)
Comment (Andrewgdewar) [reply fec.config.js:15]: This is how the moduleName was listed in my app, and the template I was given to implement. 
Please site if there is better documentation somewhere.

File: fec.config.js (hunk 7 above)
Comment (Andrewgdewar) [reply fec.config.js:8]: Will check.

File: fec.config.js (hunk 6 above)
Comment (Andrewgdewar) [reply fec.config.js:6]: We can in this case.

File: fec.config.js (hunk 8 above)
Comment (gkarat) [reply fec.config.js]: which is now resolved correct with the last commit :) resolving the thread

**Review by gkarat — APPROVED** [review]

**Comment by Andrewgdewar** [comment]
(Deployment does not have minimum availability.) << 
If this failure is unrelated to this PR, would we be able to merge this sooner rather than later?

**Comment by LightOfHeaven1994** [comment]
So the long story short - PR is fine itself but pr_check is failing on deploying `vmaas` service and patchman it set as dependency. Some other services are failing only because of `vmaas` pods can't get up. I run pr check on master and see the same issue. This one LGTM too

**Review by LightOfHeaven1994 — APPROVED** [review]
